jbi/services/jira.py

```diff
@@ -197,7 +197,7 @@ def _validate_permissions(all_projects_perms):
 
 def _all_projects_components_exist(actions: Actions):
     components_by_project = {
-        action.parameters.jira_project_key: action.parameters.jira_components
+        action.parameters.jira_project_key: action.parameters.jira_components.set_custom_components
```

Comment:
Not something to address in this PR, but:

Looking at this check again, it seems a little less valuable considering we can only check that the _custom_ components are set in the heartbeat. We don't know if all of the possible components that come via a bug will be set in the project.

I wonder if it's possible and if it would be beneficial to give the Jira automation bot the permissions to add components to projects when they don't exist. That way, we could add the component to the project before setting it on the issue if a certain component didn't exist.

Reply:
> I wonder if it's possible and if it would be beneficial to give the Jira automation bot the permissions to add components to projects when they don't exist. That way, we could add the component to the project before setting it on the issue if a certain component didn't exist.

Yes, I don't think this permission could be harmful anyway...